### DISCUSSION TITLE
[@types/node]: ChildProcess should be a class exported by child_process, not an interface

### DISCUSSION THREAD

**Discussion by AgentEnder** [DefinitelyTyped/DefinitelyTyped#55400] [Issues with a @types package]
ChildProcess is a class and works with `instanceof`, but the typings currently show it as an interface so typescript throws an error.

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
My opinion, there was no intent to use this directly. I understand the use case though.
What's your implementation use case?

**Reply by AgentEnder** [reply]
Implementation use case is checking if a returned value is a child process or not. Something similar to below:

```typescript
function executeScript(command: string, watch: boolean): Buffer | ChildProcess {
    if (watch) 
        return ... // spawn, fork etc returns ChildProcess
	else
		return execSync(command) // or otherwise return Buffer
}
```

You could theoretically use isBuffer, or some other type guard, but it still seems that `isChildProcess = x instanceOf ChildProcess` should work.

**Reply by AgentEnder** [reply]
Currently, I am using @ts-ignore on the type guard, which seems unintuitive: https://github.com/nx-dotnet/nx-dotnet/blob/master/packages/utils/src/lib/utility-functions/childprocess.ts#L5

**Reply by AgentEnder** [reply]
I went ahead and opened a PR making the swap, willing to make changes that are deemed necessary.
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55402
